### PR TITLE
Feature/groupe

### PR DESCRIPTION
**Pull request by MarcBoillot**

### CODE REVIEW

File: src/main/java/fr/ecolnum/projectapi/DTO/CandidateDto.java [hunk 1]
```diff
@@ -51,12 +52,22 @@ public CandidateDto(Candidate candidate) {
         }
     }
 
+    public Candidate convertToCandidateObject(final GroupRepository groupRepository) throws IdNotFoundException {
+        Set<Group> belongsTo = extractSetFromRepository(groupRepository, this.belongsToGroup);
+
+        return new Candidate(this.id, this.firstName, this.lastName, belongsTo);
+    }
```
Comment (aflori) [inline src/main/java/fr/ecolnum/projectapi/DTO/CandidateDto.java]: why create a new function? the Candidate model doesn't have a set of associated Pools (Set<Pool>)

File: src/main/java/fr/ecolnum/projectapi/DTO/GroupDto.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import java.util.Optional;
+import java.util.Set;
+
+import static fr.ecolnum.projectapi.util.GenericUtility.extractSetFromRepository;
+
+public class GroupDto {
+    private int id;
+    private String name;
+    private List<Integer> containedCandidates;
+    private GroupRepository groupRepository;
```
Comment (aflori) [inline src/main/java/fr/ecolnum/projectapi/DTO/GroupDto.java]: why does a DTO need a repository here?

File: src/main/java/fr/ecolnum/projectapi/DTO/GroupDto.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            for (Candidate candidateList : belongsTo) {
+                containedCandidates.add(candidateList.getId());
+            }
+        }
+    }
+    public GroupDto(int id, String name) {
+        this.id = id;
+        this.name = name;
+    }
+
```
Comment (aflori) [inline src/main/java/fr/ecolnum/projectapi/DTO/GroupDto.java:59]: conversion to model function?

File: src/main/java/fr/ecolnum/projectapi/controller/CandidateController.java [hunk 4]
```diff
@@ -66,11 +68,19 @@ public ResponseEntity<?> createCandidate(@RequestPart String firstName,
...
-     * @param lastName       lastName of checked candidate
-     * @param photoCandidate
+     *
+     * @param candidate
      * @return
      */
+    @PostMapping("/group")
+    public ResponseEntity<?> addToGroup(@RequestBody CandidateDto candidate){
+        try {
+            Candidate addedToGroup = candidateService.addToGroup(candidate);
```
Comment (aflori) [inline src/main/java/fr/ecolnum/projectapi/controller/CandidateController.java]: what group candidate is added?

File: src/main/java/fr/ecolnum/projectapi/service/CandidateService.java [hunk 5]
```diff
@@ -128,4 +132,9 @@ public Iterable<CandidateDto> returnDuplicate(CandidateDto candidate) {
 
         return duplicateCandidate;
     }
+    public Candidate addToGroup(CandidateDto candidateDto) throws IdNotFoundException{
+        Candidate candidate = candidateDto.convertToCandidateObject(groupRepository);
+        candidate = candidateRepository.save(candidate);
+        return new Candidate();
+    }
```
Comment (aflori) [inline src/main/java/fr/ecolnum/projectapi/service/CandidateService.java]: is it a change or modification of candidate?
A rename is needed?

File: src/main/java/fr/ecolnum/projectapi/controller/CandidateController.java (hunk 4 above)
Comment (aflori) [reply src/main/java/fr/ecolnum/projectapi/controller/CandidateController.java]: seeing the service, is it a modifyCandidate endpoint more than addToGroup?

File: src/main/java/fr/ecolnum/projectapi/DTO/GroupDto.java (hunk 2 above)
Comment (MarcBoillot) [reply src/main/java/fr/ecolnum/projectapi/DTO/GroupDto.java]: deleted

File: src/main/java/fr/ecolnum/projectapi/DTO/GroupDto.java (hunk 3 above)
Comment (MarcBoillot) [reply src/main/java/fr/ecolnum/projectapi/DTO/GroupDto.java:59]: not understand

Comment (MarcBoillot) [reply src/main/java/fr/ecolnum/projectapi/DTO/GroupDto.java:59]: make sentences

File: src/main/java/fr/ecolnum/projectapi/controller/CandidateController.java (hunk 4 above)
Comment (MarcBoillot) [reply src/main/java/fr/ecolnum/projectapi/controller/CandidateController.java]: isnt group of candidate just a candidate

File: src/main/java/fr/ecolnum/projectapi/DTO/CandidateDto.java (hunk 1 above)
Comment (MarcBoillot) [reply src/main/java/fr/ecolnum/projectapi/DTO/CandidateDto.java]: just say if no necessary or if i forget anythings. its more util

**Review by gillesreist — APPROVED** [review]

File: src/main/java/fr/ecolnum/projectapi/model/Group.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+import java.util.Set;
+
+@Entity
+@Table(name = "group_table")
+public class Group {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+    @Column(name = "name", nullable = false)
+    private String name;
```
Comment (aflori) [inline src/main/java/fr/ecolnum/projectapi/model/Group.java:14]: are you sure that a group cannot have a name?

File: src/main/java/fr/ecolnum/projectapi/controller/CandidateController.java [hunk 7]
```diff
@@ -5,6 +5,7 @@
 import fr.ecolnum.projectapi.exception.IdNotFoundException;
 import fr.ecolnum.projectapi.exception.MultipartFileIsNotImageException;
 import fr.ecolnum.projectapi.DTO.CandidateDto;
+import fr.ecolnum.projectapi.model.Candidate;
```
Comment (aflori) [inline src/main/java/fr/ecolnum/projectapi/controller/CandidateController.java:8]: Candidate model is used?

File: src/main/java/fr/ecolnum/projectapi/controller/PoolController.java [hunk 8]
```diff
@@ -73,6 +75,22 @@ public ResponseEntity<?> createPool(@RequestBody PoolDto pool) {
...
+    @PutMapping("/{id}/group")
+    @ApiResponse(
+            description = "Return pools list and the OK HTTP response",
+            responseCode = "201"
+    )
+    public ResponseEntity<?> addGroup(@RequestBody GroupDto groupDto, @PathVariable int id) {
+        GroupDto addedGroup;
+        try {
+            addedGroup = poolService.addGroup(groupDto, id);
+            return new ResponseEntity<>(addedGroup, HttpStatus.OK);
```
Comment (aflori) [inline src/main/java/fr/ecolnum/projectapi/controller/PoolController.java:88]: wrong file?

**Review by aflori — APPROVED** [review]

**Pull request merged**